### PR TITLE
Avoid database inconsistencies on updating problematic datasets (Connect #1689 )

### PR DESCRIPTION
**Pull request by tangrammer**
Problematic datasets are those that have merge transformations that use datasets columns that don't exist anymore, maybe the column was removed, maybe the dataset was removed

- [ ] **Update release notes if necessary**
- [x] check new queries work properly with more test data
- [X] check dependent datasets
- [x] check dependent dataset columns

### CODE REVIEW

**Comment by tangrammer** [comment]
@iperdomo perhaps you want to take a look to these SQL additions, I've tested locally the "update-dataset" UI functionality using https://github.com/akvo/akvo-lumen/tree/dev-endpoint-to-download-local-files branch 
also I've executed the new sqls against https://customer.elephantsql.com/instance dev database comparing manually the results  and they seem consistent :) 

Also I've tried the `EXPLAIN` clause with them but don't know very well how to intepret the results :!

example 

```
explain select DISTINCT ON (dataset_id) dataset_id, id, version, transformations, columns
FROM dataset_version
WHERE dataset_id IN ('1') 
order by dataset_id, version desc;
```

```
  | QUERY PLAN                                                                                        |
  |---------------------------------------------------------------------------------------------------|
  | Unique  (cost=9.51..9.52 rows=2 width=130)                                                        |
  | ->  Sort  (cost=9.51..9.52 rows=2 width=130)                                                      |
  | Sort Key: version DESC                                                                            |
  | ->  Bitmap Heap Scan on dataset_version  (cost=4.16..9.50 rows=2 width=130)                       |
  | Recheck Cond: (dataset_id = '1'::text)                                                            |
  | ->  Bitmap Index Scan on dataset_version_dataset_id_version_key  (cost=0.00..4.16 rows=2 width=0) |
  | Index Cond: (dataset_id = '1'::text)                                                              |

```

Maybe you can find more in these results :)

**Comment by tangrammer** [comment]
@kardan here you are another PR to review! :)

File: backend/src/akvo/lumen/dataset.clj [hunk 1]
```diff
@@ -3,10 +3,12 @@
   (:require [akvo.lumen.endpoint.job-execution :as job-execution]
             [akvo.lumen.import :as import]
             [akvo.lumen.lib :as lib]
+            [akvo.lumen.transformation.merge-datasets :as transformation.merge-datasets]
             [akvo.lumen.update :as update]
+            [clojure.tools.logging :as log]
             [clojure.java.jdbc :as jdbc]
             [clojure.string :as str]
-            [clojure.set :refer (rename-keys)]
+            [clojure.set :refer (rename-keys) :as set]
```
Comment (kardan) [inline backend/src/akvo/lumen/dataset.clj]: ":as set" is not needed, we're not referencing set anywhere

Comment (kardan) [inline backend/src/akvo/lumen/dataset.clj]: ":as set" is not needed, we're not referencing set anywhere

Comment (kardan) [inline backend/src/akvo/lumen/dataset.clj]: "set" is never used.

Comment (tangrammer) [reply backend/src/akvo/lumen/dataset.clj]: Updated!

**Pull request merged**